docs/cosmology/io/index.rst

```diff
@@ -0,0 +1,156 @@
+.. _cosmology_io:
+
+*********************
+Cosmology I/O/Convert
+*********************
+
+.. _cosmology_io_introduction:
+
+Introduction
+============
+
+The ``astropy.cosmology.io`` package provides a unified interface for reading, writing,
+and converting |Cosmology| objects. Most of the features of this package are accessible
+through the |Cosmology| class, which provides the methods |Cosmology.read|,
+|Cosmology.write|, |Cosmology.to_format|, and |Cosmology.from_format| for reading,
+writing, and converting |Cosmology| objects, respectively.
+
+
+Getting Started: Reading and Writing
+====================================
+
+The |Cosmology| class includes two methods, |Cosmology.read| and |Cosmology.write|, that
+make it possible to read from and write to files.
+
+The registered ``read`` / ``write`` formats include "ascii.ecsv" and "ascii.html", like
+for Table. Also, custom ``read`` / ``write`` formats may be registered into the Astropy
+Cosmology I/O framework. For more information on the built-in formats, see
+:ref:`cosmology_io_builtin`, or :ref:`cosmology_io_custom` for information on
+registering custom formats.
+
+Writing a cosmology instance requires only the file location and optionally, or if the
+file format cannot be inferred, a keyword argument "format". Additional positional
+arguments and keyword arguments are passed to the relevant writer methods.
+
+.. doctest-skip::
+
+    >>> from astropy.cosmology import Planck18
+    >>> Planck18.write("example_cosmology.ecsv", format="ascii.ecsv")
+
+
+Reading back the cosmology is done from |Cosmology|.
+
+.. doctest-skip::
+
+    >>> from astropy.cosmology import Cosmology
+    >>> cosmo = Cosmology.read("example_cosmology.ecsv", format="ascii.ecsv")
+    >>> cosmo == Planck18
+    True
+
+More specific cosmology classes can be used, providing default values for missing
+information (see :ref:`cosmology_io_details`). However using the base class is safest as
+it provides no default information and therefore requires the file to have all necessary
+information to describe a cosmology.
+
+To see a list of the available read/write file formats:
+
+.. code-block:: python
+
+    >>> from astropy.cosmology import Cosmology
+    >>> Cosmology.read.list_formats()
+      Format   Read Write Auto-identify
+    ---------- ---- ----- -------------
+    ascii.ecsv  Yes   Yes           Yes
+    ascii.html  Yes   Yes           Yes
+
+    >>> Cosmology.write.list_formats()
+       Format   Read Write Auto-identify
+    ----------- ---- ----- -------------
+     ascii.ecsv  Yes   Yes           Yes
+     ascii.html  Yes   Yes           Yes
+    ascii.latex   No   Yes           Yes
```

Comment:
@taldcroft, Adding the Write table includes ``ascii.latex``. The asymmetry between the ``read/write.list_formats()`` might be something to fix in `io.registry`.

Reply:
I'm not sure exactly what you mean. In both cases the `list_formats()` method includes a row or each available format for read or write.

Reply:
Since `ascii.latex` isn't registered with `read` it doesn't show up in `read.list_formats()`. That seems sensible, except that `read.list_formats()` also has a column for `write`! It's very confusing that the output of `read.list_formats()` and `write.list_formats()` would be different if they have columns for each other.